src/spox/_standard.py

```diff
@@ -65,13 +66,26 @@ def to_singleton_onnx_model(
         # We inject the evaluated attribute values here and then substitute back
         self_attrs = self.attrs
         try:
+            current_fields = self_attrs.get_fields().items()
```

Comment:
It would be great to keep the caching logic contained in the `_attributes.py` file. I think the previous code had the intention of "dereferencing" the `value` here. Maybe we could add a `deref` method to the `Attr` class which creates a new `Attr` instance if `self` is a `_Ref` and `self` otherwise.

Reply:
I.e. rather than calling `type(v)(v.value)` as done now, we would call `v.deref()`

Reply:
Is https://github.com/Quantco/spox/pull/123/commits/12573b5f2849ef1a8f605f0c6d0a7f76cba791fc what you had in mind?

Reply:
IIUC, calling `type(v)(v.value)` is essentially meant to deref. If so, we may add an `Attr.deref` method that returns `self` if `v._value` is not `_Ref` and otherwise returns `type(self)(self.value)`.

Reply:
The original code here was only concerned with dereferencing `_Ref`s. The way it was written, it stripped away our cache, but that was just an accident. 

It is unrelated to this line in the code, but note that we still cannot cache the Protobuf object for `AttrGraph` objects because we acutally never build it in `_attributes.py` in the first place: https://github.com/onnx/onnx/blob/main/docs/Operators.md#reshape